mlflow/utils/proto_json_utils.py

```diff
@@ -1,17 +1,81 @@
 import base64
 import datetime
 
+import json
 from json import JSONEncoder
 
 from google.protobuf.json_format import MessageToJson, ParseDict
+from google.protobuf.descriptor import FieldDescriptor
 
 from mlflow.exceptions import MlflowException
 from collections import defaultdict
+from functools import partial
+
+
+def _mark_int64_fields(proto_message):
+    """Converts a proto message to JSON, preserving only the int64/uint64/fixed64 fields."""
+
+    json_dict = {}
+    for field, value in proto_message.ListFields():
+        if field.type == FieldDescriptor.TYPE_MESSAGE:
+            ftype = partial(_mark_int64_fields)
+        elif field.type in [
+            FieldDescriptor.TYPE_INT64,
+            FieldDescriptor.TYPE_UINT64,
+            FieldDescriptor.TYPE_FIXED64,
+            FieldDescriptor.TYPE_SFIXED64,
+            FieldDescriptor.TYPE_SINT64,
+        ]:
+            ftype = int
+        else:
+            # Skip all non-int64 fields.
+            continue
+
+        json_dict[field.name] = (
+            [ftype(v) for v in value]
+            if field.label == FieldDescriptor.LABEL_REPEATED
+            else ftype(value)
+        )
+    return json_dict
+
+
+def _merge_json_dicts(from_dict, to_dict):
+    """Merges the json elements of from_dict into to_dict. Only works for json dicts
+    converted from proto messages
+    """
+
+    for key in from_dict:
+        value = to_dict[key]
+        if isinstance(value, dict):
+            _merge_json_dicts(from_dict[key], to_dict[key])
+        elif isinstance(value, list):
```

Comment:
Could value be a tuple?

Reply:
Good question! According to [this page](https://realpython.com/python-json/), Python dict constructed from a JSON string cannot have tuples:
<img width="713" alt="Screen Shot 2021-11-05 at 10 49 31" src="https://user-images.githubusercontent.com/5786126/140555891-e56a97a7-7fc6-4a0e-9788-0300319f1002.png">